sdk/keyvault/keyvault-certificates/review/keyvault-certificates.review.api.md

```diff
@@ -0,0 +1,286 @@
+    name?: string;
+}
+
+// @public
+export interface KeyProperties {
+    curve?: JsonWebKeyCurveName;
+    exportable?: boolean;
+    keySize?: number;
+    keyType?: JsonWebKeyType;
+    reuseKey?: boolean;
+}
+
+// @public
+export interface KeyVaultClientGetCertificateIssuersOptionalParams extends coreHttp.RequestOptionsBase {
+    maxresults?: number;
+}
+
+// @public
+export interface KeyVaultClientGetDeletedCertificatesOptionalParams extends coreHttp.RequestOptionsBase {
+    includePending?: boolean;
+    maxresults?: number;
+}
+
+// @public
+export interface KeyVaultClientImportCertificateOptionalParams extends coreHttp.RequestOptionsBase {
+    certificateAttributes?: CertificateAttributes;
+    // Warning: (ae-forgotten-export) The symbol "CertificatePolicy" needs to be exported by the entry point index.d.ts
+    certificatePolicy?: CertificatePolicy_2;
+    password?: string;
+    tags?: {
+        [propertyName: string]: string;
+    };
+}
+
+// @public
+export interface KeyVaultClientSetCertificateIssuerOptionalParams extends coreHttp.RequestOptionsBase {
+    attributes?: IssuerAttributes;
+    credentials?: IssuerCredentials;
+    organizationDetails?: OrganizationDetails;
+}
+
+// @public
+export interface KeyVaultClientUpdateCertificateIssuerOptionalParams extends coreHttp.RequestOptionsBase {
+    attributes?: IssuerAttributes;
+    credentials?: IssuerCredentials;
+    organizationDetails?: OrganizationDetails;
+    provider?: string;
+}
+
+// @public
+export interface KeyVaultClientUpdateCertificateOptionalParams extends coreHttp.RequestOptionsBase {
+    certificateAttributes?: CertificateAttributes;
+    certificatePolicy?: CertificatePolicy_2;
+    tags?: {
+        [propertyName: string]: string;
+    };
+}
+
+// @public
+export interface LifetimeAction {
+    // Warning: (ae-forgotten-export) The symbol "Action" needs to be exported by the entry point index.d.ts
+    action?: Action;
+    // Warning: (ae-forgotten-export) The symbol "Trigger" needs to be exported by the entry point index.d.ts
+    trigger?: Trigger;
+}
+
+// @public
+export interface NewPipelineOptions {
+    // (undocumented)
+    HTTPClient?: HttpClient;
+    // (undocumented)
+    logger?: HttpPipelineLogger;
+    // (undocumented)
+    proxyOptions?: ProxyOptions;
+    // (undocumented)
+    retryOptions?: RetryOptions;
+    telemetry?: TelemetryOptions;
+}
+
+// @public
+export interface OrganizationDetails {
+    // Warning: (ae-forgotten-export) The symbol "AdministratorDetails" needs to be exported by the entry point index.d.ts
+    adminDetails?: AdministratorDetails[];
+    id?: string;
+}
+
+// @public (undocumented)
+export interface ParsedKeyVaultEntityIdentifier {
+    name: string;
+    vaultUrl: string;
+    version?: string;
+}
+
+// @public
+export interface ProxyOptions {
+    // (undocumented)
+    proxySettings?: string;
+}
+
+// @public
+export interface RetryOptions {
+    readonly maxRetryDelayInMs?: number;
```

Comment:
I'm having a difficult time understanding the interplay between the `maxRetryDelayInMs` fields and the `retryIntervalInMs` fields.

Looking at the doc comment I think I get it (the fields are basically mutually exclusive?) which makes me think there are two use cases:

a. Choosing your own amount of time to wait between retries
or
b. Allowing the default exponential backoff policy to run and optionally setting the maxRetryDelayInMs

So could RetryOptions have a field like this?

`retryDelay: ExponentialBackoffWithMax | number;`

with `ExponentialBackoffWithMax` being something like:

interface ExponentialBackoffWithMax  {
   maxDelay?: number;
}